### ISSUE TITLE
Frog requires sentence IDs

### ISSUE THREAD

**Issue by proycon** [LanguageMachines/frog#35]
Not a priority, but for future reference: Frog can't deal with input FoLiA documents where the sentences do not have ID:

**workaround:** I created a ``foliaid`` tool in the FoLiA-tools that will generate IDs, after that Frog works fine.

```
terminate called after throwing an instance of 'folia::ValueError'
  what():  Unable to generate an id from ID= 

Program received signal SIGABRT, Aborted.
[Switching to Thread 0x2aaab2729700 (LWP 21871)]
0x00002aaaac93fc37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56      ../nptl/sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  0x00002aaaac93fc37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00002aaaac943028 in __GI_abort () at abort.c:89
#2  0x00002aaaac231535 in __gnu_cxx::__verbose_terminate_handler() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00002aaaac22f6d6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00002aaaac22e799 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00002aaaac22f34a in __gxx_personality_v0 () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00002aaaac4e4fd3 in ?? () from /lib/x86_64-linux-gnu/libgcc_s.so.1
#7  0x00002aaaac4e54f7 in _Unwind_Resume () from /lib/x86_64-linux-gnu/libgcc_s.so.1
#8  0x00002aaaaadad70e in ~FoliaElement (this=0x2aab14007210, __in_chrg=<optimized out>) at /vol/customopt/lamachine.dev/include/libfolia/folia_impl.h:72
Python Exception <class 'IndexError'> list index out of range: 
#9  folia::EntitiesLayer::EntitiesLayer (this=this@entry=0x2aab14007210, a=std::map with 1 elements, d=d@entry=0x7fffffffd480, __in_chrg=<optimized out>, __vtt_parm=<optimized out>)
    at /vol/customopt/lamachine.dev/include/libfolia/folia_impl.h:2338
#10 0x00002aaaaadabb12 in addEntity (sent=sent@entry=0x25526e0, tagset="http://ilk.uvt.nl/folia/sets/frog-ner-nl", words=std::vector of length 1, capacity 1 = {...}, 
    confs=std::vector of length 1, capacity 1 = {...}, NER="per") at ner_tagger_mod.cxx:347
#11 0x00002aaaaadac0f8 in NERTagger::addNERTags (this=this@entry=0x2aaac00008c0, words=std::vector of length 10, capacity 16 = {...}, 
    tags=std::vector of length 10, capacity 16 = {...}, confs=std::vector of length 10, capacity 16 = {...}) at ner_tagger_mod.cxx:398
#12 0x00002aaaaadaceb1 in NERTagger::Classify (this=0x2aaac00008c0, swords=std::vector of length 10, capacity 16 = {...}) at ner_tagger_mod.cxx:504
#13 0x00002aaaaad49eac in FrogAPI::TestSentence () at FrogAPI.cxx:429
#14 0x00002aaaad8ae34a in ?? () from /usr/lib/x86_64-linux-gnu/libgomp.so.1
#15 0x00002aaaac6f3184 in start_thread (arg=0x2aaab2729700) at pthread_create.c:312
#16 0x00002aaaaca06ffd in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```

**Comment by kosloot** [comment]
This is a libfolia issue. it makes ucto fail, and thus frog

**Comment by kosloot** [comment]
The libfolia problem is fixed. And also Ucto is adapted a little bit, to 'so the right thing'

**Comment by kosloot** [comment]
fixed in libfolia 1.4.3 and ucto
